dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/report-uploaded/report-uploaded.event-handler.ts

```diff
@@ -19,11 +21,28 @@ export type UtilisationReportReportUploadedEvent = BaseUtilisationReportEvent<'R
  */
 export const handleUtilisationReportReportUploadedEvent = async (
   report: UtilisationReportEntity,
-  { azureFileInfo, reportCsvData, uploadedByUserId, requestSource }: ReportUploadedEventPayload,
-): Promise<UtilisationReportEntity> =>
-  await UtilisationReportRepo.updateWithUploadDetails(report, {
-    azureFileInfo,
-    reportCsvData,
+  { azureFileInfo, reportCsvData, uploadedByUserId, requestSource, transactionEntityManager }: ReportUploadedEventPayload,
+): Promise<UtilisationReportEntity> => {
+  const azureFileInfoEntity = AzureFileInfoEntity.create({
+    ...azureFileInfo,
+    requestSource,
+  });
+  report.updateWithUploadDetails({
+    azureFileInfo: azureFileInfoEntity,
     uploadedByUserId,
     requestSource,
   });
+  await transactionEntityManager.save(UtilisationReportEntity, report);
+
+  const feeRecordEntities: FeeRecordEntity[] = reportCsvData.map((dataEntry) =>
+    feeRecordCsvRowToSqlEntity({
+      dataEntry,
+      requestSource,
+      report,
+    }),
+  );
+  await transactionEntityManager.save(FeeRecordEntity, feeRecordEntities, { chunk: 100 });
+
+  report.feeRecords = feeRecordEntities;
+  return report;
```

Comment:
I'm surprised the linter didn't get angry with this bit as well because we're modifying a function parameter. Because we have annotated the return type of this function, I think it would be fine to remove the error by returning `{ ...report, feeRecords: feeRecordEntities };`, but that's only if we think this error is worth addressing

Reply:
Just checked and it seems that the linter does get angry with this as well 😆

Reply:
I'm just going to remove line 46, right now it has no use anyway

Reply:
I think it does do something. Without that line, `report.feeRecords` would be `undefined` (or `[]` - I'm not sure which by default but I think `undefined` probably)

Reply:
Ye I suppose I was being lazy and meaning we don't use those feeRecords everywhere. Cause of typing issues I'll just add an `updateWithFeeRecords` method to the utilisationReportEntity to fix this